### PR TITLE
ci: switch flake8 by ruff

### PR DESCRIPTION
**Pull request by Borda**
Ruff is quite a new linter with impresive speed and offers to add multiple checks...

> Ruff aims to be orders of magnitude faster than alternative tools while integrating more functionality behind a single, common interface.
Ruff can be used to replace [Flake8](https://pypi.org/project/flake8/) (plus dozens of plugins), [isort](https://pypi.org/project/isort/), [pydocstyle](https://pypi.org/project/pydocstyle/), [yesqa](https://github.com/asottile/yesqa), [eradicate](https://pypi.org/project/eradicate/), [pyupgrade](https://pypi.org/project/pyupgrade/), and [autoflake](https://pypi.org/project/autoflake/), all while executing tens or hundreds of times faster than any individual tool.

Also, I would suggest installing pre-commit bot, which helps to fix code within each PR so, lowering developer overhead... see: https://pre-commit.ci/

### CODE REVIEW

**Comment by Borda** [comment]
> x **pre-commit** failed. Please run `pre-commit run --all-files` locally and commit the changes. Find more information in the repository's CONTRIBUTING.md

Also, I would suggest installing pre-commit bot, which helps to fix code within each PR, lowering developer overhead... see: [pre-commit.ci](https://pre-commit.ci/)
-> then this will be redundant: https://github.com/LAION-AI/Open-Assistant/blob/main/.github/workflows/pre-commit.yaml

**Comment by olliestanley** [comment]
Will the config you've set it up with cause it to lint with the exact same rules as `flake8` had?

**Comment by Borda** [comment]
> Will the config you've set it up with cause it to lint with the exact same rules as `flake8` had?

With my best knowledge, yes
(I have converted a few my personal projects and haven't seen any downside)

**Comment by olliestanley** [comment]
> > Will the config you've set it up with cause it to lint with the exact same rules as `flake8` had?
> 
> With my best knowledge, yes (I have converted a few my personal projects and haven't seen any downside)

Ok, in that case I am fine with this change but will wait for others to respond

**Comment by Borda** [comment]
UPDATE, with https://pre-commit.ci/#configuration-skip we can use autofix via the bot for all hooks except locals :chipmunk:

**Comment by Borda** [comment]
mind have a look @yk @andreaskoepf pls :chipmunk:

**Comment by Borda** [comment]
friendly ping @yk @andreaskoepf :otter:

**Comment by Borda** [comment]
Gentle ping @yk @andreaskoepf :chipmunk: also fine to close it :cowboy_hat_face:

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -62,16 +71,17 @@ repos:
...
-    hooks:
-      - id: flake8
-
   - repo: https://github.com/pycqa/isort
     rev: 5.12.0
     hooks:
       - id: isort
 
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.240
```
Comment (Borda) [inline .pre-commit-config.yaml:80]:
```suggestion
    rev: v0.0.263
```

**Comment by andreaskoepf** [comment]
ruff seems to be interesting, before merge someone needs to verify that it is working seamlessly as a flake8 replacement for the OA project.

**Comment by Borda** [comment]
> verify that it is working seamlessly

what do you mean? :chipmunk:

**Comment by andreaskoepf** [comment]
closed in favor of #3323 (which includes these changes, could not push merge to original repo since it was archive/is read-only now)

**Comment by Borda** [comment]
> could not push merge to original repo since it was archived)

you see me responding promptly, so you could just remind me to unarchive it :D
and wai a few hours while I was waiting for several months :flamingo:

**Pull request closed without merging**